Code/Legacy/CrySystem/Log.cpp

```diff
@@ -1281,19 +1281,17 @@ void CLog::LogStringToFile(AZStd::string_view message, ELogType logType, bool ap
         }
     }
 
-    // do not OutputDebugString in release.
+    // do not output in release.
 #if !defined(_RELEASE)
     if (queueState == MessageQueueState::NotQueued)
     {
-        if (!timeStr.empty())
-        {
-            AZ::Debug::Platform::OutputToDebugger({}, timeStr);
-        }
-        AZ::Debug::Platform::OutputToDebugger({}, message);
-        if (!message.ends_with('\n'))
-        {
-            AZ::Debug::Platform::OutputToDebugger({}, "\n");
-        }
+        AZStd::string formattedLog = AZStd::string::format(
+            "%s%s%s", 
+            !timeStr.empty() ? timeStr.data() : "", 
+            message.data(), 
+            message.ends_with('\n') ? "" : "\n");
+
+        AZ::Debug::Trace::Instance().RawOutput(nullptr, formattedLog.c_str());
```

Comment:
This function was printing the output without using `AZStd::string` to try avoid any heap allocations.

Even the [LogStringType](https://github.com/o3de/o3de/blob/6636a3a12331365eba06d63dbe902a2968f1fb15/Code/Legacy/CrySystem/Log.h#L45) variable that is being used to hold the time string is a `fixed_string` which uses stack memory.

That is the reasons, why the time string + message + (trailing newline) was output in 3 separate calls.

Reply:
> This function was printing the output without using `AZStd::string` to try avoid any heap allocations.
> 
> Even the [LogStringType](https://github.com/o3de/o3de/blob/6636a3a12331365eba06d63dbe902a2968f1fb15/Code/Legacy/CrySystem/Log.h#L45) variable that is being used to hold the time string is a `fixed_string` which uses stack memory.
> 
> That is the reasons, why the time string + message + (trailing newline) was output in 3 separate calls.

the problem is going to RawOutput is that we get:
: Something: Something :

because each call added the "<window> : <message>" line, and you get a whole lot of :'s instead.